lib/logstash/filters/fieldmap.rb

```diff
@@ -56,18 +60,61 @@ def filter(event)
       #  the key lenght, if not, explode
       split_src = event[@src_field].split(/#{@regex}/)
       @logger.debug? and @logger.debug("split_src is: ", :split_src => split_src)
+
+      if @text_qualifier
+        split_src_new_idx = split_src_orig_idx = 0
+        qualifier_start_idx = -1
+        split_src.map do |val|
+
+          val = val.strip   # this will cause problems if the text_qualifier is a space character
+
+          # Handle the case where we're in a qualified string
+          if val.start_with? @text_qualifier
```

Comment:
I think its slightly safer to flip these two conditionals, ie, process the closing qualifier before the starting qualifier.  This will help a bit with unescaped qualifiers.  ie:

'2012-03-03 01:10:01' 'Do not 'touch that dial!'

where the split regex is ' ' and the qualifier is ', there is an unescaped ' at the start of touch.  This is a contrived example and probably an error in the source system but it is a possible error we can recover from if we flip the order of checking for starting or ending of the string.